File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLAbsChatPhoto.java
package com.github.badoualy.telegram.tl.api;

import com.github.badoualy.telegram.tl.core.TLObject;

/**
 * Abstraction level for the following constructors:
 * <ul>
 * <li>{@link TLChatPhoto}: chatPhoto#6153276a</li>
 * <li>{@link TLChatPhotoEmpty}: chatPhotoEmpty#37c1011c</li>
 * </ul>
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public abstract class TLAbsChatPhoto extends TLObject {

    public TLAbsChatPhoto() {
    }

    public abstract boolean isEmpty();

    public abstract boolean isNotEmpty();

    public TLChatPhoto getAsChatPhoto() {
        return null;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLAbsChatPhoto.kt
package com.github.badoualy.telegram.tl.api

import com.github.badoualy.telegram.tl.core.TLObject

/**
 * Abstraction level for the following constructors:
 * * [chatPhoto#6153276a][TLChatPhoto]
 * * [chatPhotoEmpty#37c1011c][TLChatPhotoEmpty]
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
abstract class TLAbsChatPhoto : TLObject() {
    fun isEmpty(): Boolean = this is TLChatPhotoEmpty

    fun isNotEmpty(): Boolean = this is TLChatPhoto

    fun asChatPhoto(): TLChatPhoto? = this as? TLChatPhoto
}
